[v1.5rc][W13-B1] ContactSails

amad-person opened the pull request:
@SuxianAlicia @AJZ1995 @A0143487X

v1.5rc

AJZ1995 committed:
add colours to tag labels

novellius committed:
Week 7 Enhancement Model Component (#38)

* Added deleteTag() method in Model.java, implemented in ModelManager.java. Added dummy removeTag() method in AddressBook.java.

* Regression Testing: Implemented deleteTag() in ModelStub class in AddCommandTest.java

* Added New Exception for tags not present in master tag list. Added real removeTag() method in AddressBook.java

* Added RemoveUnusedTags() method to AddressBook.java to remove unused tags in master tag list.

* JUnit tests: Added tests to AddressBookTest.java to test removeTags() method and removeUnusedTags() which is invoked in removePerson() and updatePerson().
testUtils: Added TypicalTags.class

* Edited AddressBook.java, UniqueTagList.java, TagNotFoundException.java to follow coding standards.

* Edited AddressBookTest.java to follow coding standards.

amad-person committed:
Week 7 Enhancement: Logic Component (#37)

* Added alias words for commands

* Updated AddressBookParser to recognize command aliases

* Initial edit of UserGuide

* Updated UserGuide to include command aliases

* Added a short summary describing command aliases in UserGuide

* Added JUnit tests for testing command aliases

* Renamed undo and redo command alias JUnit tests

* Corrected Checkstyle errors

* Corrected Checkstyle errors

amad-person committed:
Week 7: Updated README (#42)

* Updated README with badges to team repository

* Updated name of application

amad-person committed:
Added User Stories and more terms in Glossary of the DeveloperGuide (#43)

amad-person committed:
Week 7: Use Cases  (#40)

* Added Use Cases to DeveloperGuide

* Improved Use Case - Export Contacts

novellius committed:
DeveloperGuide.adoc: Target User Profile, Value proposition and Non-functional Requirements. (#41)

Edited Target User Profile, Value Proposition and Non-functional requirements sections in DeveloperGuide.adoc

amad-person committed:
Updated name of application in documentation (#45)

okkhoy commented on README.adoc:
```diff
@@ -1,10 +1,10 @@
-= Address Book (Level 4)
+= ContactSails
```
:+1: You are among the few teams who has updated the title of the project :smile:

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,13 +782,19 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 *Target user profile*:
 
-* has a need to manage a significant number of contacts
+* promotes and sells products or services through social media
```
Maybe add a one-line description of what is your target user. All things listed here are the characteristics of your user.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -806,12 +812,71 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |lazy user |use short-forms for commands |execute commands more quickly
+
+|`* * *` |task-oriented user |implement meetings or deadlines for contacts |keep track of tasks to do for the contacts
```
what do you mean implement meetings? you may want to reword this one.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -837,33 +902,369 @@ Use case ends.
+*MSS*
+
+1.  User requests for current customers
+2.  ContactSails finds and displays a list of the list of customers whose orders are currently being fulfilled by user
```
There is a bit of disconnect between 1 and 2. You are requesting the current customers, but the system returns customers whose orders are being fulfilled?

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -837,33 +902,369 @@ Use case ends.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+.  The ContactSails should be usable even for users without experience in social media marketing.
```
errr... don't you think this is unrealistic? 
you are targetting for SMM, but you want your product to be used by people who have no experience in SMM?

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -837,33 +902,369 @@ Use case ends.
-_{More to be added}_
+.  The ContactSails should be usable even for users without experience in social media marketing.
+.  As ContactSails is modified from an open-source software, users should be able to download it for free.
+.  ContactSails should continue to run normally even when problems occur during execution.
```
I am not sure if this is a NFR.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -837,33 +902,369 @@ Use case ends.
+.  As ContactSails is modified from an open-source software, users should be able to download it for free.
+.  ContactSails should continue to run normally even when problems occur during execution.
+.  In case ContactSails has crashed, changes made to it during those sessions should not be lost.
+.  Size of ContactSails should not exceed 100 MB even with many contacts stored in it.
```
This is kinda unfair to the system :stuck_out_tongue_winking_eye:  you want to add a million contacts but size should be at 100mb??

okkhoy reviewed (commented):
I feel your user stories cover a broader range of applications than your intended target; maybe you want to re-strategize on what you want to deliver for your targetted users?

okkhoy commented:
@amad-person @SuxianAlicia @AJZ1995 @A0143487X
Some comments added. Please take a look and update the document as necessary.
Don't close this pull request; we will use it for further reviews later in the semester.

AJZ1995 committed:
Merge pull request #39 from AJZ1995/ui

add feature

amad-person committed:
Revert "Week 7 Enhancement UI Component" (#59)

amad-person committed:
Updated AboutUs (#54)

Added display icon and correct links

amad-person committed:
Updated UserGuide  (#58)

* Added 'Coming in v2.0' features
* Fixed phrasing in UserGuide

amad-person committed:
Updated README [Work In Progress] (#55)

* Updated README
* Updated README after PR review

amad-person committed:
Updated DeveloperGuide based on comments received on v1.0 (#57)

novellius committed:
Updated AboutUs.adoc (#63)

Updated AboutUs.adoc, removed unused pictures

novellius committed:
V1.1 Ui Update (#64)

novellius committed:
V1.1: Find by tag enhancement (#53)

Created command that allows users to find persons based on their tags.

amad-person committed:
Added autocomplete functionality for commands (#51)

* Added auto-complete functionality for commands
* Refactored CommandSyntaxListUtil
* Added GUITest for Tab keypress events
* Updated UserGuide for informing user about autocomplete functionality
* Updated UserGuide with an example of autocomplete functionality

amad-person committed:
Added autocomplete for the new FindTag command (#65)

AJZ1995 committed:
Update AboutUs and remove unused files

AJZ1995 committed:
update about us

AJZ1995 committed:
Update AboutUs [Jing Zhe] (#66)

* Update AboutUs and remove unused files

A0143487X committed:
Update AboutUs [Kevin] (#67)

AJZ1995 committed:
Update file name

AJZ1995 committed:
Merge branch 'master' into AboutUs

AJZ1995 committed:
Merge pull request #68 from AJZ1995/AboutUs

Updated AboutUs

novellius committed:
V1.2 Classification of Tag into Group and Preference (#70)

* Created Subclasses for Tag: Group and Preference to represent different categories of tags for a Person.

* Edited Ui to accommodate new labels for preferences and groups

* Refactored FindTagCommand to FindGroupCommand and created FindPreferenceCommand

* Added methods to Model API allow deleting of group and preference.

amad-person committed:
Changed picture and updated UserGuide (#69)

okkhoy reviewed (commented):
I can see user guide updates only to some features. So, I have consider you missed the milestone!

okkhoy commented:
@amad-person @AJZ1995 @SuxianAlicia @A0143487X 

I don't see user guide updates to most of the things on this list:

+* Manage customer contacts: Add, edit and delete contacts.
+* Categorize customers: Group customers according to their preferences, social media platforms, or any other category you want to define.
+* Manage orders: Add, edit, and delete customer orders.
+* Send promotions: Apply differentiated marketing strategies by sending promotions to relevant customers.
+* Keep track of deadlines: Add tasks and reminders and view your schedules in the integrated calendar.
+* Know your customer demographic: See what the top selling items and who your most frequent customers are.

For those you want to implement as a part of later milestones, you were supposed to think through and write the command format and usage examples now!
The idea is to get you to think through the product from an end user perspective, and develop it accordingly.


In addition,
I would want you to mention the following in your **Developer guide**
1. For each team member what are the features (major and minor) you are proposing?
2. Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

amad-person commented:
@okkhoy In the 2103 website, the TODO just mentioned to mark the `coming in v2.0` features. We added a section in the UserGuide that does this. Also, in the Developer Guide, we put the release versions in the User Stories table. Can this please be considered as passing the milestone? Thanks!

amad-person committed:
Added Feature Contributions in DevGuide Appendix B (#74)

Description of Major and Minor Features added, with a summary of how these features fit into the product scope. [Aadyaa]

amad-person committed:
Orders: Implemented ability to add orders to ContactSails (#52)

* Added Orders to application Model
* Implemented AddOrderCommand to add orders
* Added AddOrderCommandParser and ParserUtil logic
* AddressBookParser recognizes AddOrderCommand and executes it
* Added JUnit tests

novellius committed:
Developer Guide Feature Contribution [Alicia] (#76)

* Added Feature Contribution to DeveloperGuide.adoc.

AJZ1995 committed:
Added Feature Contributions to Developer Guide [Jing Zhe] (#82)

A0143487X committed:
Added Feature Contribution to Developer Guide [Kevin]

amad-person committed:
V1.2: Updated User Guide for OrderCommands

* Updated User Guide for OrderCommands

amad-person committed:
V1.2: Update Developer Guide [Aadyaa] 

* Orders
* Encryption

novellius committed:
V1.2 Documentation Update [Alicia] (#87)

* Updated Implementation in Developer Guide to include Calendar and Event tracking feature with relevant diagrams.
Updated User Guide to include calendar and event management features.

novellius committed:
V1.2 Delete groups and preference Commands (#89)

* Created DeleteGroupCommand, DeletePreferenceCommand and their respective parsers.

* Added tests for new Commands.

* Edited function headers.

* Updated documentation for the new commands.

* Added more test cases.

amad-person committed:
Updated UI Mockup with actual screenshot

amad-person committed:
V1.2: Delete orders, Storage and UI

* Added OrderListCards
* Added OrderListPanel for viewing a list of orders in GUI
* Added DeleteOrderCommand
* Added JUnit tests
* Added XmlAdaptedOrder
* Added XML JUnit tests
* Updated Use Cases in DevGuide for Order operations

novellius committed:
Added System Tests for FindGroup and FindPreference Commands. (#90)

* Added System Tests for the Find Group and Find Preference Commands.

okkhoy commented on docs/UserGuide.adoc:
```diff
@@ -204,61 +379,243 @@ Examples:
+=== Clearing all entries in Event List : `clearevent` `Coming in V1.5`
+
+Clears all event list entries from the address book. +
+Format: `clearevent` +
```
Good to see substantial updates to the UG :+1:

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -358,13 +358,290 @@ image::UndoRedoActivityDiagram.png[width="650"]
+----
+public class Order {
+
+    private final OrderInformation orderInformation;
```
Please use code snippets minimally.
Reason: it is difficult to maintain these in the document, particularly if the implementation changes.

okkhoy reviewed (commented):
I would say generally good updates to DG.
Please use code snippets minimally; use UML diagrams to explain your implementation. 
For some of the commands/features I noticed that you haven't given the design considerations. Please update the same.

okkhoy commented:
@amad-person @SuxianAlicia @AJZ1995 @A0143487X
Some comments added to the DG. Please take a look and update as necessary.

amad-person committed:
Update Developer Guide (#101)

* Updated DeveloperGuide
* Updated links to other .adoc documents

novellius committed:
DevGuide Documentation Update [DG Draft1] (#102)

* Edited Appendices.

* Edited DevOps section.

* Edited Implementation Section 3.5

novellius committed:
Changed Syntax for group and preference related commands. (#103)

novellius committed:
DevGuide Update (#104)

amad-person committed:
Updated Design section in Developer Guide

amad-person committed:
V1.3: Change themes

* Added ChangeThemeCommand
* Added JUnit tests for ChangeThemeCommand and its parser
* Updated User Guide for the ChangeTheme command

novellius committed:
System Tests for Delete Group and Delete Preference. (#106)

amad-person committed:
V1.3: Edit Orders 

* Added EditOrderCommand
* Added JUnit tests
* Added GUI tests for OrderListPanel

amad-person committed:
Changed version of application to v1.3

novellius committed:
V1.3: Calendar Update

* UI: Created CentrePanel to facilitate switching between CalendarPanel and BrowserPanel
* Logic: ViewCalendarCommand, AddCalendarCommand
* Model: Created CalendarEvent class in Model and list to store multiple events
* Storage: Created XmlAdaptedCalendarEvent for storage purposes, and updated XmlSerializable AddressBook to maintain XmlAdaptedCalendarEvents.
* Testing: Added tests for the new classes, and fixed some of the existing tests.

amad-person committed:
Collated code [Aadyaa]

amad-person committed:
Added collated files [Aadyaa]

AJZ1995 committed:
V1.4: Added PersonPanel 

- Added PersonPanel 
- Added and fixed JUnit tests

amad-person committed:
V1.4: Updated application version

amad-person committed:
V1.4: Change Order Status

* Added ChangeOrderStatusCommand
* UI updates according to ChangeOrderStatusEvent

amad-person committed:
V1.4: Updated UserGuide

* Information about new commands added
* Coming in v2.0 commands added

amad-person committed:
V1.4: Updated mockup and improved UI

AJZ1995 committed:
Collated code [Jing Zhe]

novellius committed:
Made empty cells of list view transparent. (#137)

novellius committed:
Changed Calendar to CalendarFX. (#133)

Replaced existing Calendar with CalendarFX.
Created panel to show existing calendar entries in ContactSails.
Created commands to allow switching between the entry list panel and order list panel.
Added DeleteEntryCommand to remove Entries by their Index in Entry listing.
Modified ViewCalendarCommand to allow users to view Calendar in Month, Day and Week.
Modified UG to reflect changes.
Collated Code.

novellius committed:
Updated UG. (#140)

Added missing command in CommandSyntaxListUtil.java

amad-person committed:
Updated UserGuide

amad-person committed:
Fixed HelpWindow bug

amad-person committed:
Updated invalid format message in AddOrderCommand

amad-person committed:
Updated application name in Config.java

amad-person committed:
Updated LightTheme

amad-person committed:
Documentation Fix: UserGuide

* Fixed AddCommand example
* Categorized commands into sections
* Revamped Command Summary section
* Formatted tables in Command Summary section

novellius committed:
DG Update (#173)

* Updated Current Implementation for Calendar Interface and entry tracking.

* Updated AddEntry sequence diagram.

* Updated CalendarEntryClassDiagram

* Updated Model and Storage Class Diagram

novellius committed:
Fixed Command Example. (#177)

novellius committed:
Changed Initial Display size of CentrePanel, Calendar Panel and corresponding pane in MainWindow. (#176)

novellius committed:
Fix Person Panel not updating information when Undo/Redo/Clear is executed. (#179)

* Fixed Person Panel showing outdated person information when undo command is executed, by reseting person panel display.

* Fixed person panel display when clear is executed.

novellius committed:
New command to edit existing calendar entries. (#181)

* Refactored name of AddEventCommandParser to AddEntryCommandParser

* Created Utility class for checking time-related fields of Calendar Entry.

* Added methods required to update existing entries in calendar manager.

* Changed AddEntryCommandParser and XmlAdaptedCalendarEntry to use functions in new Utility class.

* Created EditEntryCommand to allow users to edit existing entries.

* Added EditEntryCommand to CommandSyntaxListUtil and AddressBookParser

* Edited UG in parts related to EditEntryCommand.

* Fixed bug where after updating calendar entry in Calendar Manager, calendar does not reflect new change.

* Created tests for EditEntryCommand and EditEntryCommandParser.

* Implemented updateCalendarEntry() in ModelStubs in tests.

* Fixed AddEntryCommandParser to use new Utils class to check calendar entry's time-related attributes.

* Edited existing utility classes. Fixed Bug in checking for entry duration in EntryTimeConstraintsUtil.

* Added Tests for Utility Classes DateUtil, TimeUtil, and EntryTimeConstraintsUtil. Added EditEntryDescriptorTest.

* Modified assertCommandFailure in CommandTestUtil. 

* Edited and added more test cases within AddressBookParserTest.

amad-person committed:
V1.5rc: New Application icon (#188)

* Updated Application icon
* Reduced icon size to fix tests

amad-person committed:
Bug Fix: Select Command  (#187)

Fixes #180.

AJZ1995 committed:
Bug Fix: Multiple Tags in PersonPanel (#185)

novellius committed:
Calendar Buttons CLI Command Equivalent Update (#190)

* Created the 3 commands to view today, previous page, next page of calendar.

* Modifed CentrePanel to listen to ChangeCalendarPageRequestEvent. Added corresponding methods to handle the request in CalendarPanel.

* Added the new commands to AddressBookParser and CommandSyntaxListUtil.

* Changed location of collate annotations.

* Refactored variable names and updated UG.

* Edited ViewCalendarCommand.

* Created Tests for ViewCalendarCommand, ViewBackCommand, ViewNextCommand and ViewTodayCommand.

novellius committed:
Fix Button Text Color in DarkTheme.css and made list-view cells transparent in LightTheme.css. (#193)

* Fixed Color of text of buttons of Calendar.

* Made ListView empty cells transparent in LightTheme.css

AJZ1995 committed:
V1.5rc: Add Tag colours (#186)

AJZ1995 committed:
Minor language updates in Developer Guide

amad-person committed:
Bug Fix: Updated Price and Quantity Constraints (#192)

* Added more robust constraints for Price and Quantity fields in Orders

AJZ1995 committed:
Project Portfolio Page [Jing Zhe] (#194)

novellius committed:
Added Project Portfolio Page. (#184)

amad-person committed:
V1.5rc: Update Orders (#196)

* Added OrderStatus to storage and updated OrderStatus command

amad-person committed:
Project Portfolio Page [Aadyaa] (#197)

* Updated ProjectPortfolio and rebuilt collated code

* Linked ProjectPortfolio to AboutUs

AJZ1995 committed:
minor language changes

AJZ1995 committed:
Merge pull request #195 from AJZ1995/developerGuide

Minor language updates in Developer Guide

novellius committed:
Updated UG Section 3 (#200)

* Added images to be used in ContactSails Ui explanation.

* Updated section 3 on Ui of ContactSails.

amad-person committed:
Added glossary to UserGuide (#201)

AJZ1995 committed:
Collated Code [Jing Zhe] (#202)

amad-person committed:
V1.5rc: Add icons for fields (#191)

* Improved UI
* Fixed blank PersonPanel bug
* Refactor icon images path
* Reduced size of all images

novellius committed:
UserGuide: Fixed links to broken images. (#206)

* Fixed Broken links in UG.

* Updated Pictures used in UG.

* Updated images used in UG.

novellius committed:
Bug fix: Selecting selected person while calendar is displayed will not display Person Panel. (#208)

* Refactored name of CentrePanel to CenterPanel, ResetPersonPanelEvent ro ResetPersonPanelRequestEvent.

* Added DisplayPersonPanelRequestEvent to display person panel via select command.

* Further refactoring of name [centrePlaceholder to centrePlaceholder]

* Updated UG description on Calendar Command.

AJZ1995 committed:
Bug Fix: Lack of Tag Colour and Update of Person Card to Person Panel (#209)

novellius committed:
Calendar Jump Update (#210)

* Added CalendarJumpCommand and its parser. Added new Event to facilitate switching of displayed calendar date.

* Added Command to AddressBookParser and CommandSyntaxUtilList. Added method in ParserUtil to parse date and added prefix for TARGET_DATE. Added message to be used in CalendarJumpCommandParser.

* Edited CentrePanel to subscribe to new Event, and added necessary method in CalendarPanel to switch date.

* Changed event posted in AddEntry and EditEntry Command, to post new event such that the date in calendar will be switched to display the new/edited entries.

* Corrected messages in EndTime and StartTime.

* Updated UG.

* Fixed Example mistake.

* Changed Class name in AddEntryCommandTest

* Added test cases to AddressBookParserTest, added new tests CalendarJumpCommandTest and CalendarJumpCommandParser Test. Added TypicalLocalDates used in these tests.

* Refactored name of DisplayCalendarRequestEvent to ChangeCalendarViewRequestEvent.

amad-person committed:
UserGuide Update (#212)

* Added pictures for commands in the Managing Orders section.
* Tweaked the Setting Up and Using ContactSails sections.
* Updated Command Summary to include new commands added.

novellius committed:
UserGuide Update: Calendar and Calendar Entries Section (#213)

* Added images used in Calendar and Calendar Entries sections.
* Edited Calendar and Calendar Entries section in UG.
* Edited Glossary section in UG.
* Edited Command Summary in UG.

amad-person committed:
Removed AppVeyor badge and updated UI screenshot (#223)

novellius committed:
Calendar Panel Update (#222)

* Changed name of calendarPanelholder to calendarPanelHolder in CalendarPanel.java and CalendarPanel.fxml.

* Removed unused methods in CenterPanel.

* Created TypicalBaseEvents used in tests.

* Added CenterPanelHandle and edited CalendarPanelHandle to be used in tests.

* Created CalendarPanelTest.

* Refactored CalendarUtil to CalendarEntryConversionUtil.

* Updated files related to the refactoring of CalendarUtil to CalendarEntryConversionUtil.

* Extracted constants used in managing the calendar, such as those changing calendar view, and requests to go to the next page, previous page and today's page. Constants are placed in CalendarFxUtil.

* Shifted Collated annotation in TimeUtil.

* Cleared unused imports in CenterPanel, CenterPanelHandle and CalendarPanelHandle.

* Fixed coding violation in CalendarManager.

* Added annotation to CalendarPanelTest.

amad-person committed:
Orders: Refactored and Updated Tests (#221)

* Refactored and updated tests for Order model
* Added ChangeOrderStatusCommand tests

novellius committed:
Calendar Entry List Clear Update (#224)

* Created EntryListClearCommand.

* Added EntryListClearCommand to AddressBookParser and CommandSyntacListUtil.

* Created Test for EntryListClearCommand.

* Updated UG.

* Changed description of ClearCommand.

* Added test cases related to EntryListClearCommand in AddressBookParserTest.

AJZ1995 committed:
Updated User Guide: Managing Persons and Bug Warning  (#215)

AJZ1995 committed:
Updated Developer Guide : Person Panel Section (#214)

amad-person committed:
Standardise UserGuide (#217)

novellius committed:
Conform code to Coding Standard (#225)

* Removed empty lines.

* Changed way of calling EntryTimeContraintsUtil.checkCalendarEntryTimeConstraints().

* Edited CalendarManager Class Javadoc Comment.

* Edited message of fail() in checkCalendarEntryTimeConstraints_validInputs_success() of EntryTimeConstraintsUtilTest.

* Edited comments in CalendarEntryCardTest.

* Changed reference to static member in CalendarEntryListPanelTest to correct Class instance.

* Removed extra spaces and relocated annotations.

* Corrected expectedModel changes in command tests that integrates with Model.

* Removed unused imports.

* Collated code.

* Renamed package event to entry.

* Fixed method names in DeleteGroupCommandTest and DeletePreferenceCommandTest.

* Changed comment of assertCommandFailure method in CommandTestUtil.

* Corrected equals() method in StartTime and EndTime.

* Corrected method names, constant names, comments.

* Added spacing for StartDateTest and StartTimeTest.

* Added 1 additional test case to XmlUtilTest.

* Added annotation to test case in XmlUtilTest.

* Removed unneeded spaces and lines.

* Corrected comments and method names in DeleteGroupCommandTest, FindGroupCommandTest, and FindPreferenceCommandTest.

* Added assertions to tests to check instanceof BaseEvents.

* Corrected line length for XmlUtilTest.

A0143487X committed:
Export Command (#199)

Added export command

To do:
Look at improving to command workflow
Update UGDG
Add test cases

Reference:
* Added Kevin Yeong photo, Proj portfolio and section in AboutUs
* Added export to CSV file function with exception handling
* Updated UG for export to CSV function
Edited files to match style guide.
* Update UserGuide.adoc
* Update ExportToCsvCommand.java
* Update ExportToCsvCommandParser.java
* Update ExportToCsvCommandParser.java

amad-person committed:
Update DevGuide and Project Portfolio (#226)

* Added and updated diagrams
* Updated implementation sections for major and minor features
* Updated setting up section by adding pictures
* Updated feature contribution section in appendix
* Revamped project portfolio

AJZ1995 committed:
Update Project Portfolio [Jing Zhe] (#229)

amad-person committed:
Final CleanUp (#227)

* Updated Documentation
* Updated Project Portfolio
* Code clean up for functional and test code
* Collated code

novellius committed:
Address book SystemTest Update (#231)

* Changed name of rightPlaceholder in RightPanel to rightPlaceHolder.

* Added tests for RightPanel.

* Removed unused varables and parameters in CalendarPanelTest and CenterPanelHandle.

* Edited MainWindowHandle to provide accessors to OrderListPanelHandle, CalendarEntryListPanelHandle, PersonPanelHandle, and CalendarPanelHandle.

* Widened window size in TestApp so that HelpWindowTest can click on RightPanel nodes.

* Refactored name of method in GuiTestAssert.

* Edited assertApplicationDisplaysExpected in AddressBookSystemTest.

* Removed unused import in CenterPanelHandle.

* Added methods in GuiTestAssert and AddressBookSystemTest to check whether OrderListPanel and CalendarEntryListPanel display the correct orders and calendar entries in given Model respectively.

* Edited Order-related Command System Tests to test whether Order List Panel displays the same list as Model's filtered Order List.

* Refactored name of method in Model from updateFilteredCalendarEventList() to updateFilteredCalendarEntryList.

* Updated CommandTestUtil constants related to CalendarEntry.

* Updated constants used in Entry-related tests.

* Corrected constant description in EntryTitle.

* Added methods to be used in Entry-related System Tests.

* Edited comments in System Tests.

* Added AddEntry, DeleteEntry and EditEntry System Tests.

* Corrected merge errors.

* Changed UniqueOrderList to use List instead of Set. Removed methods to check that UniqueOrderList and UniqueCalendarEntryList is order-insensitive.

* Removed unused import in UniqueCalendarEntryList.

* Refactored DuplicateOrderException to exception package within order package.

* Commented out problematic test.

A0143487X committed:
Export: Tests and Documentation (#230)

amad-person committed:
Recollate code - final.

amad-person committed:
Updated broken images and added note about undoable commands

amad-person committed:
Reduced size of images in orderdelete section

amad-person committed:
UserGuide Fix

- fix broken images
- add section for undoable commands

novellius committed:
Sample Data [CalendarEntry] Update (#235)

* Added sample calendar entries.

* Corrected indentation.

novellius committed:
Fixed description at 5.7.1 of UG

amad-person committed:
Added sample data for orders (#237)

novellius committed:
DG Final Update (#238)

* Updated Undo/Redo Section in DG.

* Edited DG, PPP.

* Updated Section 4.7.

* Edited Appendix H. Corrected message in EntryTimeConstraintsUtil.

* Updated DG.

* Added and edited existing Diagrams.

* Corrected Figure Numbers

* Updated PPP.

* no message
